test/e2e/aws.go

```diff
@@ -83,12 +100,11 @@ func (a *awsTestHelper) ensureHostedZone(rootDomain string) (string, []string, e
 	if err != nil {
 		return "", nil, fmt.Errorf("failed to get create hosted zone: %w", err)
 	}
-	nameServers = aws.StringValueSlice(zone.DelegationSet.NameServers)
-	hostedZoneID = aws.StringValue(zone.HostedZone.Id)
-	return hostedZoneID, nameServers, nil
+	return aws.StringValue(zone.HostedZone.Id), aws.StringValueSlice(zone.DelegationSet.NameServers), nil
 }
 
-func (a *awsTestHelper) deleteHostedZone(zoneID string) error {
+// AWS sdk expect to zone ID to delete the xone, where as Azure expect Domain Name
+func (a *awsTestHelper) deleteHostedZone(zoneID, rootDomain string) error {
```

Comment:
```suggestion
func (a *awsTestHelper) deleteHostedZone(zoneID, zoneDomain string) error {
```
Purely cosmetic: to be consistent with the previous comment.

Reply:
Done